File: selenium/src/test/java/com/ytfc/blog/pages/user/LoginPage.java
package com.ytfc.blog.pages.user;

import com.ytfc.blog.utility.Helper;
import com.ytfc.blog.forms.LoginForm;
import com.ytfc.blog.pages.PageObject;

import org.openqa.selenium.WebDriver;

public class LoginPage extends PageObject {
    private LoginForm form;

    public LoginForm getForm() {
        return this.form;
    }

    public LoginPage(WebDriver driver) {
        this.driver = driver;
        this.form = new LoginForm(driver);
    }

    public LoginPage get() {
        Helper.get(driver, "user/login");

        assertPageTitle("Login", "Page has correct title");
        form.assertFieldTypes("Field has correct type");

        return this; // Chainable
    }

    public LoginPage login(String reqName, String reqPassword) {
        get();
        form.getField("name").setValue(reqName);
        form.getField("password").setValue(reqPassword);
        form.getButton().click();

        return this; // Chainable
    }

}

File: cypress/cypress/pages/login-page.js
import { Common } from '../pages/common'
import { LoginForm } from '../forms/login-form'

export class LoginPage extends Common {
    constructor() {
        super()
        this._form = new LoginForm()
    }

    visit() {
        cy.visit('/user/login')
        return this
    }

    login(name, password) {
        this.getForm().enter(new Map([['name', name], ['password', password]]))

        return this
    }


    clickRegisterLink() {
        this.visit()
        cy.contains('Register new account').click()
    }

    clickForgotNameLink() {
        this.visit()
        cy.contains('Forgot user name').click()
    }

    clickForgotPasswordLink() {
        this.visit()
        cy.contains('Forgot password').click()
    }
}
